### ISSUE TITLE
hierdec-trio_16bar trained model hierarchical decoder has one unused RNN level

### ISSUE THREAD

**Issue by ChromaticIsobar** [tensorflow/magenta#1677]
The `hierdec-trio_16bar` trained model has a `HierarchicalLstmDecoder` decoder.

The number of levels `num_levels` of a `HierarchicalLstmDecoder` is defined in `HierarchicalLstmDecoder._hierarchical_decode` as `len(self._level_lengths) - 1`.

RNN levels are listed in `self._hier_cells`: `self._hier_cells[i]` is the RNN for level `i`. The list is accessed only for `0 <= i < num_levels`.

In the `hierdec-trio_16bar` trained model
 - `self._level_lengths` is `[16, 16]`
 - so, `num_levels` (`= len(self._level_lengths) - 1`) is 1
 - so, only `self._hier_cells[0]` is used
 - `self._hier_cells` is a list of 2 `tensorflow.python.ops.rnn_cell_impl.MultiRNNCell`
 - so, `self._hier_cells[1]` is never used

Actively deleting the level (`del self._hier_cells[-1]`) did not result in any error, even when sampling from the decoder. This enforces my belief that the last level is never used.

Although this is not a critical issue, removing the level might yield a reduction of the object size.

**Comment by adarob** [comment]
Thanks Marco. Are you saying you can delete the level and still load a pre-trained checkpoint?

**Comment by ChromaticIsobar** [comment]
Yes, I can delete a level from the hierarchical decoder pre-trained checkpoint (I'm using the `hierdec-trio_16bar` pre-trained checkpoint) and get the same output from the decoder using the same `z`.

I call the method that has the delete instruction at the beginning of the `_hierarchical_decode` method, so that I am absolutely sure that the level is deleted before the graph for the output is built (deleting both levels does result in an error).

**Comment by adarob** [comment]
Thanks for testing. Do you want to make a PR to remove the building of it?

**Comment by ChromaticIsobar** [comment]
If you think it could be good, I will do it when I'll have the time.
